label_sync/main.go

```diff
@@ -186,22 +198,43 @@ func validate(labels []Label, parent string, seen map[string]string) error {
 	return nil
 }
 
+func copyStringMap(originalMap map[string]string) map[string]string {
+	newMap := make(map[string]string)
+	for k, v := range originalMap {
+		newMap[k] = v
+	}
+	return newMap
+}
+
 // Ensures the config does not duplicate label names
 func (c Configuration) validate() error {
+	// Check default labels
 	seen := make(map[string]string)
-	if err := validate(c.Labels, "", seen); err != nil {
+	if err := validate(c.Default, "default", seen); err != nil {
 		return fmt.Errorf("invalid config: %v", err)
 	}
+	// Check other repos labels
+	for repo, labels := range c.Repos {
+		// Duplicate seen to avoid modifications
+		defaultLabels := copyStringMap(seen)
+		// Validate as usual
+		// Will complain if a label is both in default and repo
+		if err := validate(labels, repo, defaultLabels); err != nil {
```

Comment:
The `seen` map in `validate()` ends up being used in two ways and you wrote the `copyStringMap()` to work around a side-effect as well -- we should refactor so there is an unambiguous API where you get a `seen` map back and an api where you pass one in to use, etc

Reply:
I still need to make it a pure function...